clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -942,7 +942,8 @@ void EmitAssemblyHelper::EmitAssembly(BackendAction Action,
     PerModulePasses.add(createDeadArgEliminationSYCLPass());
 
   if (LangOpts.SYCLIsDevice && LangOpts.SYCLExplicitSIMD)
-    PerModulePasses.add(createGenXSPIRVWriterAdaptorPass());
+    PerModulePasses.add(
+        createGenXSPIRVWriterAdaptorPass(/*RewriteTypes=*/true));
```

Comment:
I am not entirely sure what this change does.   Can you provide some details in the commit description?
Please add an appropriate test as well.

Reply:
Yes, it would be great to have some context here. Although I'm not sure if front-end test is required here. I think this thing enables some functionality of the `GenXSPIRVWriterAdaptorPass` which is likely already tested by LIT suite for this particular pass.

Reply:
Sure, I'll add some details in the next iteration of this change. We've realized that an additional change in vc-intrinsics (library where GenXSPIRVAdaptorPass comes from) is required. So, I mark the pull request as a draft for now.